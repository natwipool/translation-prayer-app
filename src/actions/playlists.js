import uuid from 'uuid';

// ADD_PLAYLIST
export const addPlaylist = ({ name = '', lists = [] } = {}) => ({
  type: 'ADD_PLAYLIST',
  playlist: {
    id: uuid(),
    name,
    lists
  }
});

// REMOVE_PLAYLIST
export const removePlaylist = ({ id } = {}) => ({
  type: 'REMOVE_PLAYLIST',
  id
});

// EDIT_PLAYLIST
export const editPlaylist = (id, updates) => ({
  type: 'EDIT_PLAYLIST',
  id,
  updates
});
