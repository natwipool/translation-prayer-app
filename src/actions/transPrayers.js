// SET_TRANS_PRAYER
const setTransPrayer = (transPrayers = {}) => ({
  type: 'SET_TRANS_PRAYER',
  transPrayers
});

export default setTransPrayer;