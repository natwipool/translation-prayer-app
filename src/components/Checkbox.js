import React from 'react';

// class Checkbox extends React.Component {
//   constructor(props) {
//     super(props)

//     this.state = {
//       checked: props.lists.find((list) => list === props.precept)
//     }
//   }

//   onAddLists = (list) => {
//     this.setState(() => ({
//       checked: !this.state.checked
//     }))
//     console.log(this.state.checked)
//     this.props.onAddLists(list)
//   }
  
//   onRemoveLists = (list) => {
//     this.setState(() => ({
//       checked: !this.state.checked
//     }))
//     this.props.onRemoveLists(list); 
//   }

//   render() {
//     return (
//       <div>
//         <input
//           type="checkbox"
//           name={this.props.precept}
//           value={this.props.precept}
//           checked={this.state.checked}
//           onChange={e => {
//             const list = e.target.value;
//             if (e.target.checked) {
//               this.onAddLists(list)  
//             } else if (!e.target.checked) {
//               this.onRemoveLists(list);  
//             }
//           }}
//         />
//         <label htmlFor={this.props.precept}>
//           {this.props.precept}
//           {
//             this.props.lists.indexOf(this.props.precept) > -1 && (
//               this.props.lists.indexOf(this.props.precept) + 1
//             )
//           }
//         </label>
//       </div>
//     )
//   }
// }

const Checkbox = (props) => (
  <div>
    <input
      type="checkbox"
      name={props.precept}
      value={props.precept}
      checked={props.lists.find((list) => list === props.precept)}
      onChange={e => {
        const list = e.target.value;
        if (e.target.checked) {
          props.onAddLists(list)  
        } else if (!e.target.checked) {
          props.onRemoveLists(list);  
        }
      }}
    />
    <label htmlFor={props.precept}>
      {props.precept}
      {
        props.lists.indexOf(props.precept) > -1 && (
          props.lists.indexOf(props.precept) + 1
        )
      }
    </label>
  </div>
);

export default Checkbox;
