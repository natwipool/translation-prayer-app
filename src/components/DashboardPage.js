import React from 'react';
import PrefacePage from './PrefacePage'

const DashboardPage = () => (
  <div>
    <PrefacePage />
  </div>
);

export default DashboardPage;