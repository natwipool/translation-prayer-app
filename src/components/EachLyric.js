import React from 'react';

const EachLyric = (props) => (
  <div>
    {props.lyric}
  </div>
);

export default EachLyric;