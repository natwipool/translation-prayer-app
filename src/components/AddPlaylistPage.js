import React from 'react';
import { connect } from 'react-redux';
import PlaylistForm from './PlaylistForm';
import { addPlaylist } from '../actions/playlists';

export class AddPlaylistPage extends React.Component {
  addPlaylist = playlist => {
    this.props.addPlaylist(playlist);
    this.props.history.push('/playlists');
  };
  render() {
    return (
      <div>
        <h1>เพิ่มรายการสวดมนต์</h1>
        <PlaylistForm onSubmit={this.addPlaylist} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addPlaylist: playlist => dispatch(addPlaylist(playlist))
});

export default connect(undefined, mapDispatchToProps)(AddPlaylistPage);
