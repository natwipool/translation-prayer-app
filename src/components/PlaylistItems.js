import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PlaylistItem from './PlaylistItem';
import Lyrics from './Lyrics';
import Player from './Player';
import { removePlaylist } from '../actions/playlists';

export class PlaylistItems extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      playingImg: false,
      index: 0
    }

    this.transPrayerList = this.props.playlist.lists.map(list =>
      this.props.transPrayers.find(trans => list === trans.precept)
    );
  }
  handlePlayNext = () => {
    this.setState((prevState) => ({
      index: prevState.index + 1
    }))
  }
  handlePlayPrev = () => {
    this.setState((prevState) => ({
      index: prevState.index - 1
    }))
  }
  handlePlayButton = () => {
    this.setState(() => ({
      playingImg: true
    }))
  }
  handleStopButton = () => {
    this.setState(() => ({
      playingImg: false
    }))
  }
  onPlayEachPrecept = playIndex => {
    this.child.playEachPrecept(playIndex);
    this.setState(() => ({
      playingImg: true,
      index: playIndex
    }))
  };
  render() {
    return (
      <div>
        {
          this.state.playingImg && (
            <Lyrics
              index={this.state.index}
              transPrayers={this.transPrayerList}
            />
          )
        }
        <div>
          <h1>{this.props.playlist.name}</h1>
          <Link to={`/edit/${this.props.playlist.id}`}>
            <button>
              แก้ไข
            </button>  
          </Link>
          <button onClick={() => {
            this.props.dispatch(removePlaylist({ id: this.props.playlist.id }));
            this.props.history.push('/playlists')
          }}>
            ลบ
          </button>
        </div>
        {this.props.playlist.lists.map((list, index) => (
          <div key={index}>
            <PlaylistItem
              transPrayerList={this.props.transPrayers.find(
                trans => list === trans.precept
              )}
            />
            <button
              onClick={e => {
                this.onPlayEachPrecept(index);
              }}
            >
              {this.state.playingImg && this.state.index === index ? 'PLAYING' : 'PLAY'}
            </button>
          </div>          
        ))}
        <Player
          playlists={this.transPrayerList}
          handlePlayNext={this.handlePlayNext}
          handlePlayPrev={this.handlePlayPrev}
          handlePlayButton={this.handlePlayButton}
          handleStopButton={this.handleStopButton}
          onRef={ref => (this.child = ref)}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    transPrayers: state.transPrayers,
    playlist: state.playlists.find(
      playlist => playlist.id === props.match.params.id
    )
  };
};

export default connect(mapStateToProps)(PlaylistItems);
