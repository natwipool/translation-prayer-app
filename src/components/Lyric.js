import React from 'react';
import EachLyric from './EachLyric';

const Lyric = (props) => {
  return (
    <div>
      <h3>{props.transPrayer.precept}</h3>
      {props.transPrayer.lyrics.map((lyric, index) => 
        <EachLyric 
          key={index}
          lyric={lyric}
        />
      )}
    </div>
  )
}

export default Lyric;