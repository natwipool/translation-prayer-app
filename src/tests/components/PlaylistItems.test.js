import React from 'react';
import { shallow } from 'enzyme';
import { PlaylistItems } from '../../components/PlaylistItems';
import transPrayers from '../fixtures/transPrayers';
import playlists from '../fixtures/playlists';

test('should render PlaylistItems correctly', () => {
  const wrapper = shallow(<PlaylistItems transPrayers={transPrayers} playlist={playlists[2]} />);
  expect(wrapper).toMatchSnapshot();
});