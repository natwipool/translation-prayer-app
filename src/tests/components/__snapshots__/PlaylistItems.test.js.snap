// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlaylistItems correctly 1`] = `
<div>
  <div>
    <h1>
      Playlist name three
    </h1>
    <Link
      replace={false}
      to="/edit/3"
    >
      <button>
        แก้ไข
      </button>
    </Link>
    <button
      onClick={[Function]}
    >
      ลบ
    </button>
  </div>
  <div
    key="0"
  >
    <PlaylistItem />
    <button
      onClick={[Function]}
    >
      PLAY
    </button>
  </div>
  <div
    key="1"
  >
    <PlaylistItem />
    <button
      onClick={[Function]}
    >
      PLAY
    </button>
  </div>
  <div
    key="2"
  >
    <PlaylistItem />
    <button
      onClick={[Function]}
    >
      PLAY
    </button>
  </div>
  <Player
    handlePlayButton={[Function]}
    handlePlayNext={[Function]}
    handlePlayPrev={[Function]}
    handleStopButton={[Function]}
    onRef={[Function]}
    playlists={
      Array [
        undefined,
        undefined,
        undefined,
      ]
    }
  />
</div>
`;
